source/nodejs/adaptivecards-designer/src/device-emulation.ts

```diff
@@ -3,11 +3,11 @@
 
 export abstract class DeviceEmulation {
   readonly name: string;
-  readonly styleSheet: string;
+  readonly maxWidth: string | null;
```

Comment:
Any reason why `maxWidth` is a `string` and not a `number`? Only a width expressed in pixels is meaningful in the context of this feature.

Reply:
I didn't want to proscribe the use of CSS units other than `px`; I figure developers can choose what they like, in case there is some more useful value to them than a CSS pixel value.

Is there a reason this should be limited to just `px` values?

Reply:
It's just that in real life a screen width is expressed in pixels, and the idea with this feature is that the tool matches real life screens.

Reply:
I would agree if it weren't for [CSS pixels having not much bearing on hardware pixels](https://juiceboxinteractive.com/blog/a-pixel-is-not-a-pixel-designing-for-a-new-generation-of-mobile-devices/#:~:text=A%20hardware%20pixel%20is%20an%20individual%20dot%20of,a%20concept%20known%20as%20the%20device%20pixel%20ratio.).

As an example, I have an iPhone 13 Pro, and on Apple's [tech specs page](https://www.apple.com/iphone-13-pro/specs/) they say the screen is "2532‑by‑1170-pixel resolution at 460 ppi".

If I visit whatismyscreenresolution.net, however, it reports my iPhone's CSS pixel dimensions as 390x844.

I propose developers should put the whole CSS string as this value as a reminder that the width of the card is being determined by CSS and not some other graphics package, and that its `px` comes with its caveats and also that other units are available (developers could, if they like, use `in`, which happens to always be 96 times larger than `px` and thus potentially less ambiguous).

Reply:
All that said, since you have a better handle on the norms in this project, I'll gladly defer to your choice, just wanted to contextualize my choices here.

Reply:
The width of the card isn't being defined by CSS, especially not on Teams mobile which this feature is meant to help with. On mobile, cards are rendered natively (as is the whole app) and CSS isn't even involved.

Resolutions as reported by design systems and programming APIs are usually brought back to 96dpi so that developers don't have to deal with dpi conversions unless they want/need to.

To me, the questions are:
- What are the values that make sense to the end user in that drop down
- How do we translate these values into an actual width in the designer

To the first question, I'd say the values that make most sense are likely "iPhone 6", "iPhone 13", "iPad mini", "iPad Pro 10'", "Galaxy S10", etc more than actual pixel dimensions.

To the second question, "iPhone 13" as you mention above translates into 390 pixels (at 96 dpi), never to other units. That's why I think only pixels make sense: no other unit will actually be used to specify those dimensions.

All that said, it doesn't fundamentally matter that much, and if you feel strongly (as you apparently do) about width expressed as a string, I suppose we can live with it.